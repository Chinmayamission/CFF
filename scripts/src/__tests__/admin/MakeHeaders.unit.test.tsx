import { IDataOptionView } from "../../admin/FormEdit/FormEdit.d";
import Headers from "../../admin/util/Headers";

it('renders default columns for main view in response table', () => {
    const dataOptionView: IDataOptionView = {
        "id": "main",
        "displayName": "Main"
    };
    const schema = {
        "type": "object",
        "properties": {
            "name": {
                "type": "object",
                "properties": {
                    "first": { "type": "string" },
                    "last": { "type": "string" }
                }
            },
            "parents": {
                "type": "object",
                "properties": {
                    "age": { "type": "number" }
                }
            }
        }
    };
    const result = Headers.makeHeadersFromDataOption(dataOptionView, schema);
    expect(result).toMatchSnapshot();
});

it('renders custom columns for main view in response table', () => {
    const dataOptionView: IDataOptionView = {
        "id": "main",
        "displayName": "Main",
        "columns": [
            { "label": "First Names", "value": "parents.name.first" }
        ]
    };
    const schema = {
        "type": "object",
        "properties": {
            "name": {
                "type": "object",
                "properties": {
                    "first": { "type": "string" },
                    "last": { "type": "string" }
                }
            },
            "parents": {
                "type": "object",
                "properties": {
                    "age": { "type": "number" }
                }
            }
        }
    };
    const result = Headers.makeHeadersFromDataOption(dataOptionView, schema);
    expect(result).toMatchSnapshot();
});