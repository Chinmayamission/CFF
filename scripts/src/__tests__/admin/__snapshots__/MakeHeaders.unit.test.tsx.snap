// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom columns for main view in response table 1`] = `
Array [
  Object {
    "Cell": [Function],
    "Header": "First Names",
    "accessor": [Function],
    "id": "parents.name.first",
    "key": "parents.name.first",
    "label": "First Names",
  },
]
`;

exports[`renders custom columns for unwind by view in response table 1`] = `
Array [
  Object {
    "Cell": [Function],
    "Header": "First Names",
    "accessor": [Function],
    "id": "parents.name.first",
    "key": "parents.name.first",
    "label": "First Names",
  },
  Object {
    "Cell": [Function],
    "Header": "Name.last",
    "accessor": [Function],
    "id": "name.last",
    "key": "name.last",
    "label": "Name.last",
  },
]
`;

exports[`renders default columns for main view in response table 1`] = `
Array [
  Object {
    "Cell": [Function],
    "Header": "ID",
    "accessor": [Function],
    "id": "ID",
    "key": "ID",
    "label": "ID",
  },
  Object {
    "Cell": [Function],
    "Filter": [Function],
    "Header": "PAID",
    "accessor": [Function],
    "filterMethod": [Function],
    "id": "PAID",
    "key": "PAID",
    "label": "PAID",
  },
  Object {
    "Cell": [Function],
    "Header": "DATE_CREATED",
    "accessor": [Function],
    "id": "DATE_CREATED",
    "key": "DATE_CREATED",
    "label": "DATE_CREATED",
  },
  Object {
    "Cell": [Function],
    "Header": "Name.first",
    "accessor": [Function],
    "id": "name.first",
    "key": "name.first",
    "label": "Name.first",
  },
  Object {
    "Cell": [Function],
    "Header": "Name.last",
    "accessor": [Function],
    "id": "name.last",
    "key": "name.last",
    "label": "Name.last",
  },
  Object {
    "Cell": [Function],
    "Header": "Parents.age",
    "accessor": [Function],
    "id": "parents.age",
    "key": "parents.age",
    "label": "Parents.age",
  },
]
`;
